sdk/typescript/src/cryptography/ed25519-keypair.ts

```diff
@@ -79,6 +81,49 @@ export class Ed25519Keypair implements Keypair {
     return new Ed25519Keypair(nacl.sign.keyPair.fromSeed(seed));
   }
 
+  /**
+   * Test Derive Path from path string
+   *
+   * @param path path string (`m/44'/784'/0'/0'/0'`)
+   */
+  static isValidPath = (path: string): boolean => {
+    if (
+      !new RegExp("^m\\/44+'\\/784+'\\/[0-9]+'\\/[0-9]+'\\/[0-9]+'+$").test(
+        path
+      )
+    ) {
+      return false;
+    }
+    return !path
+      .split('/')
+      .slice(1)
+      .map((val: string): string => val.replace("'", ''))
+      .some(isNaN as any);
+  };
+
+  /**
+   * Generate a keypair from mnemonics.
+   *
+   * @param path path string (`m/44'/784'/0'/0'/0'`)
+   * @param mnemonics: mnemonics is a word seed phrase
+   */
+  static fromDerivePath(path: string, mnemonics: string): Ed25519Keypair {
+    if (!Ed25519Keypair.isValidPath(path)) {
+      throw new Error('Invalid derivation path');
+    }
+
+    const normalizeMnemonics = mnemonics
+      .trim()
+      .split(/\s+/)
+      .map(part => part.toLowerCase())
+      .join(' ');
+
+    const seed = bip39.mnemonicToSeed(normalizeMnemonics);
+    const { key } = derivePath(path, seed.toString('hex'));
```

Comment:
seems like you can just use `mnemonicToSeedHex` instead of `toString('hex')` https://github.com/vacuumlabs/bip39-light#examples

Reply:
Yes, I'll update with `bip39-light.d.ts`

```typescript
// Copyright (c) 2022, Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

declare module 'bip39-light' {
    export function entropyToMnemonic(
        entropy: Buffer | string,
        wordlist?: string[]
    ): string;
    export function generateMnemonic(
        strength?: number,
        rng?: (size: number) => Buffer,
        wordlist?: string[]
    ): string;
    export function mnemonicToSeed(mnemonic: string, password?: string): Buffer;
    export function mnemonicToSeedHex(mnemonic: string, password?: string): string;
    export function validateMnemonic(mnemonic: string): boolean;
}
```